gramps/gen/db/generic.py

```diff
@@ -621,6 +623,18 @@ def _initialize(self, directory, username, password):
         """
         raise NotImplementedError
 
+    def upgrade_table_for_json_access(self, table_name):
+        """
+        Overload this method to add JSON access
+        """
+        raise NotImplementedError
+
+    def supports_json_access(self):
```

Comment:
```suggestion
    def uses_json_data(self):
```
I wonder if `uses_json_data` is better name than `supports_json_access`?
My thinking: SQLite supports JSON data. Older (blob) gramps databases in SQLite *support* JSON, but don't **use** JSON.
Using similar logic, `upgrade_table_for_json_access` would become `upgrade_table_for_json_data`